libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFAction.m

```diff
@@ -541,8 +541,8 @@ - (id)contentFromData:(NSData*)data fromResponse:(NSHTTPURLResponse*)response er
         content = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:&jsonParseError];
     }
     
-    // If it's an error here, it's a basic parsing error.
-    if (jsonParseError && error) {
+    // If it's an error here, it's a basic parsing error and error is not 500 internal server error.
+    if (jsonParseError && error && response.statusCode != 500) {
```

Comment:
Oh I see, it's supposed to fall through.  It seems to me that a better approach would be to institute classes of error handling here, like `CSFSalesforceAction` has.  For instance, maybe all non-4xx errors fall through the default JSON parser, rather than being parsed.  The parsing is predicated on the idea that the REST APIs will send back error responses for bad requests, and that's probably not going to happen with a non-4xx response code.